server/graphql/v2/mutation/EmojiReactionMutations.js

```diff
@@ -0,0 +1,130 @@
+import { GraphQLNonNull, GraphQLString } from 'graphql';
+
+import { mustBeLoggedInTo } from '../../../lib/auth';
+import models from '../../../models';
+import { canComment } from '../../common/expenses';
+import { Forbidden, NotFound, Unauthorized, ValidationFailed } from '../../errors';
+import { idDecode, IDENTIFIER_TYPES } from '../identifiers';
+import { CommentReferenceInput } from '../input/CommentReferenceInput';
+import { UpdateReferenceInput } from '../input/UpdateReferenceInput';
+import { EmojiReactionsOutput } from '../object/EmojiReactionsOutput';
+
+const emojiReactionMutations = {
+  addEmojiReaction: {
+    type: new GraphQLNonNull(EmojiReactionsOutput),
+    args: {
+      emoji: {
+        type: new GraphQLNonNull(GraphQLString),
+        description: 'The emoji associated with the reaction',
+      },
+      comment: {
+        type: CommentReferenceInput,
+        description: 'A unique identifier for the comment id associated with this reaction',
+      },
+      update: {
+        type: UpdateReferenceInput,
+        description: 'A unique identifier for the update id associated with this reaction',
+      },
+    },
+    resolve: async (entity, args, req) => {
+      if (!req.remoteUser) {
+        throw new Unauthorized();
+      }
+
+      if (!args.comment && !args.update) {
+        throw new Error('A comment or update must be provided');
+      }
+
+      if (args.comment) {
+        const commentId = idDecode(args.comment.id, IDENTIFIER_TYPES.COMMENT);
+        const comment = await models.Comment.findByPk(commentId);
+
+        if (!comment) {
+          throw new ValidationFailed('This comment does not exist');
+        } else if (comment.ExpenseId) {
+          const expense = await models.Expense.findByPk(comment.ExpenseId);
+          if (!expense || !(await canComment(req, expense))) {
+            throw new Forbidden('You are not allowed to comment or add reactions on this expense');
+          }
+        }
+        const reaction = await models.EmojiReaction.addReactionOnComment(req.remoteUser, commentId, args.emoji);
+        return { comment: models.Comment.findByPk(reaction.CommentId), update: null };
```

Comment:
Already loaded above

```suggestion
        return { comment, update: null };
```

Reply:
@Betree : Yes that's correct but note that we add the reaction to the comment on the preceding line. So we need to get the updated comment, isn't? 🤔 😄

Reply:
Same here, I don't think the comment gets mutated in the operation

Reply:
Yep, you are right. I've corrected this. 😄